lib/compiler/src/engine/artifact.rs

```diff
@@ -571,7 +571,7 @@ impl Artifact {
         let mut obj = get_object_for_target(target_triple).map_err(to_compile_error)?;
 
         let object_name = format!(
-            "WASMER_METADATA_{}",
+            "WASMER_{}_METADATA",
```

Comment:
This should be computed using the Symbol Registry, not hardcoding it here

Reply:
`SymbolRegistry.symbol_to_name(wasmer_types::Symbol::Metadata);`

Reply:
Sorry, missed this when replacing.

Done in https://github.com/wasmerio/wasmer/pull/3430/commits/bf37e63bbaea7d6d0b0196c3229cf182a511aea8